diktat-maven-plugin/src/main/kotlin/org/cqfn/diktat/plugin/maven/DiktatBaseMojo.kt

```diff
@@ -237,10 +234,9 @@ abstract class DiktatBaseMojo : AbstractMojo() {
         val text = file.readText()
         val params =
             KtLint.ExperimentalParams(
-                fileName = file.relativeTo(mavenProject.basedir).path,
+                fileName = file.path,
                 text = text,
                 ruleSets = ruleSets,
-                userData = mapOf("file_path" to file.path),
```

Comment:
userData is prohibited in ktlink:
```kotlin
userData
  .keys
  .minus(editorConfigProperties)
  .let {
      check(it.isEmpty()) {
          "UserData contains properties ${it.sorted()}. However, userData is deprecated and will be " +
              "removed in a future version. Please create an issue that shows how this field is " +
              "actively used."
      }
  }
```

Reply:
Well, maybe we'll need to reopen https://github.com/pinterest/ktlint/issues/1133. Did you find any other way to access the full file path with new ktlint, when we are not using `Ktlint.lint` directly?

Reply:
`fileName` is full path (as I got):
- javadocs: `@param fileName path of file to lint/format`
- method in KtLint:
```kotlin
internal val normalizedFilePath: Path?
    get() = if (fileName == STDIN_FILE || fileName == null) {
        null
    } else {
        Paths.get(fileName)
    }
```

Reply:
`ktlint` puts `fileName` (by calling `normalizedFilePath()`) to `userData` with key `com.pinterest.ktlint.core.KtLint#FILE_PATH_PROPERTY = "file_path"`. After it, it's re-put to `com.pinterest.ktlint.core.KtLint#getFILE_PATH_USER_DATA_KEY = "FILE_PATH"`.
We use the latest one key to get a path to file in `org.cqfn.diktat.ruleset.utils.AstNodeUtilsKt#getFilePath` (we use it to calculate real package name for validation)

So, we control that `fileName` contains a full path: we put value to this property in `ExperimentalParams`